src/Workspaces/Core/Portable/Classification/SyntaxClassification/AbstractSyntaxClassificationService.Worker.cs

```diff
@@ -56,14 +57,14 @@ internal partial class AbstractSyntaxClassificationService
 
             internal static void Classify(
                 SemanticModel semanticModel,
-                TextSpan textSpan,
+                TextSpan[] textSpans,
                 SegmentedList<ClassifiedSpan> list,
                 Func<SyntaxNode, ImmutableArray<ISyntaxClassifier>> getNodeClassifiers,
                 Func<SyntaxToken, ImmutableArray<ISyntaxClassifier>> getTokenClassifiers,
                 ClassificationOptions options,
                 CancellationToken cancellationToken)
             {
-                using var worker = new Worker(semanticModel, textSpan, list, getNodeClassifiers, getTokenClassifiers, options, cancellationToken);
+                using var worker = new Worker(semanticModel, textSpans, list, getNodeClassifiers, getTokenClassifiers, options, cancellationToken);
```

Comment:
i actually think you do *not* want to do it this way.  INstead, loop over textSpans *here*, and just call the normal 'single text span' code that already exists from that point.